src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -282,15 +319,9 @@ private unsafe void RemoveNodeFromDatabase(TransactionOperationContext context,
                 }
                 remove.UpdateDatabaseRecord(databaseRecord, index);
 
-                if (databaseRecord.Topology.AllNodes.Any() == false)
+                if (databaseRecord.DeletionInProgress.Any() == false)
```

Comment:
And if we added some additional nodes here? This can only happen if both all nodes and deletion in progress are empty